tests/returntypes/test_vjp_new.py

```diff
@@ -0,0 +1,665 @@
+
+    @pytest.mark.autograd
+    def test_autograd(self, tol):
+        """Tests that the output of the VJP transform
+        can be differentiated using autograd."""
+        dev = qml.device("default.qubit.autograd", wires=2)
+        params = np.array([0.543, -0.654], requires_grad=True)
+
+        def cost_fn(x, dy):
+            with qml.tape.QuantumTape() as tape:
+                ansatz(x[0], x[1])
+
+            tape.trainable_params = {0, 1}
+            tapes, fn = qml.gradients.vjp(tape, dy, param_shift)
+            vjp = fn(dev.batch_execute(tapes))
+            return vjp
+
+        dy = np.array([-1.0, 0.0, 0.0, 1.0], requires_grad=False)
+        res = cost_fn(params, dy)
+        assert np.allclose(res, expected(params), atol=tol, rtol=0)
+
+        res = qml.jacobian(cost_fn)(params, dy)
+        assert np.allclose(res, qml.jacobian(expected)(params), atol=tol, rtol=0)
+
+    @pytest.mark.torch
+    def test_torch(self, tol):
+        """Tests that the output of the VJP transform
+        can be differentiated using Torch."""
+        import torch
+
+        dev = qml.device("default.qubit", wires=2)
+
+        params_np = np.array([0.543, -0.654], requires_grad=True)
+        params = torch.tensor(params_np, requires_grad=True, dtype=torch.float64)
+        dy = torch.tensor([-1.0, 0.0, 0.0, 1.0], dtype=torch.float64)
+
+        with qml.tape.QuantumTape() as tape:
+            ansatz(params[0], params[1])
+
+        tape.trainable_params = {0, 1}
+        tapes, fn = qml.gradients.vjp(tape, dy, param_shift)
+        vjp = fn(qml.execute(tapes, dev, qml.gradients.param_shift, interface="torch"))
+        print(vjp)
+        assert np.allclose(vjp.detach(), expected(params.detach()), atol=tol, rtol=0)
+
+        cost = vjp[0]
+        cost.backward()
+
+        exp = qml.jacobian(lambda x: expected(x)[0])(params_np)
+        assert np.allclose(params.grad, exp, atol=tol, rtol=0)
+
+    @pytest.mark.tf
+    @pytest.mark.slow
+    def test_tf(self, tol):
+        """Tests that the output of the VJP transform
+        can be differentiated using TF."""
+        import tensorflow as tf
+
+        dev = qml.device("default.qubit.tf", wires=2)
+
+        params_np = np.array([0.543, -0.654], requires_grad=True)
+        params = tf.Variable(params_np, dtype=tf.float64)
+        dy = tf.constant([-1.0, 0.0, 0.0, 1.0], dtype=tf.float64)
+
+        with tf.GradientTape() as t:
+            with qml.tape.QuantumTape() as tape:
+                ansatz(params[0], params[1])
+
+            tape.trainable_params = {0, 1}
+            tapes, fn = qml.gradients.vjp(tape, dy, param_shift)
+            vjp = fn(dev.batch_execute(tapes))
+
+        assert np.allclose(vjp, expected(params), atol=tol, rtol=0)
+
+        res = t.jacobian(vjp, params)
+        assert np.allclose(res, qml.jacobian(expected)(params_np), atol=tol, rtol=0)
+
+    # TODO: to be added when lighting and TF compatible with return types
```

Comment:
We should make a note for this in one of our shared docs.

Reply:
Yes we have too come back they is some points in the TODO list in the tests